Assets/HoloToolkit/Speech/Scripts/TextToSpeechManager.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            // Make sure there's something to speak
+            if (speakFunc == null) throw new ArgumentNullException(nameof(speakFunc));
+
+            if (synthesizer != null)
+            {
+                try
+                {
+                    // Need await, so most of this will be run as a new Task in its own thread.
+                    // This is good since it frees up Unity to keep running anyway.
+                    Task.Run(async () =>
+                    {
+                        // Speak and get stream
+                        var speechStream = await speakFunc();
+
+                        // Get the size of the original stream
+                        var size = speechStream.Size;
+
+                        // Create buffer
+                        byte[] buffer = new byte[(int)size];
+
+                        // Get input stream and the size of the original stream
+                        using (var inputStream = speechStream.GetInputStreamAt(0))
+                        {
+                            // Close the original speech stream to free up memory
+                            speechStream.Dispose();
+
+                            // Create a new data reader off the input stream
+                            using (var dataReader = new DataReader(inputStream))
+                            {
+                                // Load all bytes into the reader
+                                await dataReader.LoadAsync((uint)size);
+
+                                // Copy from reader into buffer
+                                dataReader.ReadBytes(buffer);
+                            }
+                        }
+
+                        // Load buffer as a WAV file
+                        var wav = new Wav(buffer);
+
+                        // The remainder must be done back on Unity's main thread
+                        UnityEngine.WSA.Application.InvokeOnAppThread(() =>
+                        {
+                            // Convert to an audio clip
+                            var clip = wav.ToClip("Speech");
+
+                            // Set the source on the audio clip
+                            audioSource.clip = clip;
+
+                            // Play audio
+                            audioSource.Play();
+                        }, false);
+                    });
+                }
+                catch (Exception ex)
+                {
+                    Debug.LogErrorFormat("Speech generation problem: \"{0}\"", ex.Message);
+                }
+            }
+            else
+            {
+                Debug.LogErrorFormat("Speech not initialized. \"{0}\"", text);
+            }
+        }
+        #endif
+
+        private void StartSpeech()
+        {
+            try
+            {
+                if (audioSource == null) { throw new InvalidOperationException("An AudioSource is required and should be assigned to 'Audio Source' in the inspector."); }
```

Comment:
Are you trying to throw this out of the method or just use this as a means to log the error in your catch below?